### DISCUSSION TITLE
Adding to Global "Right Click" Popup Menu

### DISCUSSION THREAD

**Discussion by rstone-ws** [NationalSecurityAgency/ghidra#5990] [Q&A] [answered]
I'm writing a plugin that needs to access different "state" aspects (mainly, currentLocation, currentAddress, etc), and I'm hoping to trigger a function accessing that information from a right-click menu within the code browser. However, I have not managed to find a way to add my plugin options to the right click "menu". Is it possible to add to the list of options that pops up when right-clicking in the "Listing" window and trigger a function within my plugin? 
So far, I've managed to set an action if I right-click inside the plugin window, but haven't found a way to add to the more "global" options in the listing window. (i.e. instruction info, patch instruction, etc options). 
Any insight/tips would be appreciated!

**Comment by ryanmkurtz** [comment, marked as answer]
This is how I recently did it:
https://github.com/NationalSecurityAgency/ghidra/blob/c225fac124821d789b12e4c76f4a0f6eabe83544/Ghidra/Features/FileFormats/src/main/java/ghidra/macosx/plugins/DyldCacheBuilderPlugin.java#L72-L87

**Reply by rstone-ws** [reply]
Super helpful!! Thanks so much, this worked.
